File: src/main/java/org/blitzortung/android/app/helper/ViewHelper.java
package org.blitzortung.android.app.helper;

import android.util.DisplayMetrics;
import android.view.View;

public final class ViewHelper {
    
    private ViewHelper() {}

    public static float pxFromSp(View view, float sp) {
        final DisplayMetrics displayMetrics = view.getContext().getResources().getDisplayMetrics();
        return sp * displayMetrics.scaledDensity;
    }

    public static float pxFromDp(View view, float dp)
    {
        final DisplayMetrics displayMetrics = view.getContext().getResources().getDisplayMetrics();
        return dp * displayMetrics.density;
    }
}

File: app/src/main/java/org/blitzortung/android/app/helper/ViewHelper.kt
/*

   Copyright 2015 Andreas Würl

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.

*/

package org.blitzortung.android.app.helper

import android.view.View

object ViewHelper {

    fun pxFromSp(view: View, sp: Float): Float {
        val displayMetrics = view.context.resources.displayMetrics
        return sp * displayMetrics.scaledDensity
    }

    fun pxFromDp(view: View, dp: Float): Float {
        val displayMetrics = view.context.resources.displayMetrics
        return dp * displayMetrics.density
    }
}
